docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -73,14 +76,40 @@ The valid values are:
 Capping replicated table size
 -----------------------------
 
-Join distribution type will be chosen automatically when join reordering strategy
-is set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``.
-In such case it is possible to cap the maximum size of replicated table via
-``join-max-broadcast-table-size`` config property (e.g. ``join-max-broadcast-table-size=100MB``)
-or via ``join_max_broadcast_table_size`` session property (e.g. ``set session join_max_broadcast_table_size='100MB';``)
-This allows to improve cluster concurrency and to prevent bad plans when CBO misestimates size of joined tables.
+Join distribution type is automatically chosen when join reordering strategy is
+set to ``AUTOMATIC`` or when join distribution type is set to ``AUTOMATIC``. In
+such cases, it is possible to cap the maximum size of the replicated table via
+the ``join-max-broadcast-table-size`` configuration property (such as
+``join-max-broadcast-table-size=100MB``) or via the
+``join_max_broadcast_table_size`` session property (such as
+``set session join_max_broadcast_table_size='100MB';``). This allows you to
+improve cluster concurrency and prevent bad plans when CBO misestimates size
+of joined tables.
+
+By default, replicated table size is capped to 100MB.
+
+Syntactic join order
+--------------------
+
+If not using cost-based optimization, Trino defaults to syntactic join ordering.
+While there is no formal way to optimize queries for this case, it is possible
+to take advantage of how Trino implements joins to make them more performant.
+
+Trino uses in-memory hash joins. When processing a join statement, Trino loads
+the right-most table of the join into memory as the build side, then streams the
```

Comment:
do we need to have an example query .. so we can explain what right-most table even means?